selene_sdk/update_seqweaver.py

```diff
@@ -0,0 +1,152 @@
+"""
+This module provides the `UpdateSeqweaver` class, which wraps the master bed file
+containing all of the features' binding sites parsed from CLIP-seq.
+It supports new dataset construction and training for Seqweaver.
+
+"""
+import h5py
+import gzip
+import numpy as np
+import sys
+
+from sequences.genome import Genome
+from targets.genomic_features import GenomicFeatures
+from samplers.dataloader import H5DataLoader
+from train_model import TrainModel
+from utils.config import load_path
+from utils.config_utils import parse_configs_and_run
+
+class UpdateSeqweaver():
+    """
+    Stores a dataset specifying sequence regions and features.
+    Accepts a tabix-indexed `*.bed` file with the following columns,
+    in order:
+        [chrom, start, end, strand, feature]
+
+    Parameters
+    ----------
+    input_path : str
+        Path to the tabix-indexed dataset. Note that for the file to
+        be tabix-indexed, it must have been compressed with `bgzip`.
+        Thus, `input_path` should be a `*.gz` file with a
+        corresponding `*.tbi` file in the same directory.
+    output_path : str
+        Path to the output constructed-training data file.
+    feature_path : str
+        Path to a '\n'-delimited .txt file containing feature names.
+    hg_fasta : str
+        Path to an indexed FASTA file -- a `*.fasta` file with
+        a corresponding `*.fai` file in the same directory. This file
+        should contain the target organism's genome sequence.
+
+    """
+    def __init__(self, input_path, train_path, validate_path, feature_path, hg_fasta, yaml_path, VAL_PROP=0.01, sequence_len=1000):
+        """
+        Constructs a new `UpdateSeqweaver` object.
+        """
+        self.input_path = input_path
+        self.train_path = train_path
+        self.validate_path = validate_path
+        self.feature_path = feature_path
+        self.yaml_path = yaml_path
+        self.VAL_PROP = VAL_PROP
```

Comment:
why is this in all caps, out of curiosity? is this some code convention you've seen before?

Reply:
Good point - I usually put global variables in all caps, which this variable temporarily was during testing but isn't anymore. Changed to lower caps :)